### PR TITLE
WEBUI-1511:Own Code Static Scan : Open Redirect

### PR DESCRIPTION
**Pull request by alokhyland**

### CODE REVIEW

**Review by rahuljain-dev — DISMISSED** [review]

File: elements/routing.js [hunk 1]
```diff
@@ -213,7 +226,8 @@ app.router = {
     }
     const isFullpath = /^http(s)?:\/\//.test(path);
     if (isFullpath) {
-      const { encodepath, isvalidUrl } = getTrustedDomains(path);
+      const isvalidUrl = isTrustedDomain(path);
```
Comment (madhurkulshrestha-hyland) [inline elements/routing.js]: rectify camel casing

File: elements/routing.js [hunk 2]
```diff
@@ -213,7 +226,8 @@ app.router = {
...
+      const encodepath = encodeQueryParams(path);
```
Comment (madhurkulshrestha-hyland) [inline elements/routing.js]: Move encoding after checking isValidUrl

File: elements/routing.js [hunk 3]
```diff
@@ -27,21 +27,34 @@ function scrollToTop(ctx, next) {
...
+  str = wwwRegex.test(str) ? str : str.replace(/^(https?:\/\/)?/, '$1www.');
+  return str;
+}
+
+function isTrustedDomain(path) {
   const trustedDomains = Nuxeo && Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.trustedDomains;
-  const pathUrl = new URL(path);
+  if (!trustedDomains) return true;
+  const modifiedPathUrl = createUrlFromString(path);
+  const pathUrl = new URL(modifiedPathUrl);
```
Comment (madhurkulshrestha-hyland) [inline elements/routing.js:42]: where is this pathurl and hostname being used ? Line 47 is again initializing hostname

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

File: elements/routing.js (hunk 3 above)
Comment (alokhyland) [reply elements/routing.js:42]: done

File: elements/routing.js (hunk 1 above)
Comment (alokhyland) [reply elements/routing.js]: done

**Review by rahuljain-dev — APPROVED** [review]

**Review by swarnadipa-dev — APPROVED** [review]

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Pull request merged**